src/core/qgsrange.h

```diff
@@ -574,26 +570,26 @@ class QgsTemporalRange
      */
     bool overlaps( const QgsTemporalRange<T> &other ) const
     {
-      if ( !mUpper.isValid() && ( ( mIncludeLower && mLower <= other.mUpper ) || ( !mIncludeLower && mLower < other.mUpper ) ) )
+      if ( !mUpper.isValid() && ( ( mIncludeLower && other.mIncludeUpper && mLower <= other.mUpper ) || ( ( !mIncludeLower || !other.mIncludeUpper ) && mLower < other.mUpper ) ) )
```

Comment:
Shouldn't this method be synchronized with QgsRange::overlaps logic?

Reply:
In fact, going by QgsRange::overlaps new logic, I found that QgsTemporalRange::overlaps was not considering infinite bounds (invalid QDate()) so we had

```python
r1 = QgsDateRange(QDate(), QDate(2000, 1, 1))
r2 = QgsDateRange(QDate(3000, 1, 1), QDate())
r1.overlaps(r2)
>> True
```

Now with the new logic we are good (hopefully) with all the combinations I could add to the tests.

Thanks for the review!